ssl/quic/quic_impl.c

```diff
@@ -4623,6 +4633,260 @@ static QUIC_CONNECTION *create_qc_from_incoming_conn(QUIC_LISTENER *ql, QUIC_CHA
     return NULL;
 }
 
+DEFINE_LHASH_OF_EX(QUIC_TOKEN);
+
+struct ssl_token_store_st {
+    LHASH_OF(QUIC_TOKEN) *cache;
+    CRYPTO_REF_COUNT references;
+    CRYPTO_MUTEX *mutex;
+};
+
+static uint64_t fnv1a_hash_token(uint8_t *key, size_t len)
```

Comment:
Re-creating this comment because the other was marked as resolved.

Looks like we already have this exact function defined here: [master/crypto/hashtable/hashtable.c#L89](https://github.com/openssl/openssl/blob/master/crypto/hashtable/hashtable.c?rgh-link-date=2025-01-31T05%3A58%3A05Z#L89).
Can we make that function non-static and use that instead?

Reply:
Hmm, maybe.  hashtable.c is an internal api that is part of libcrypto, not libssl, so even if we did a shared source implementation, we would be getting two object copies of the function.  I can try but it might be better to, in a separate PR, include the entire hashtable api in libssl, and convert the tokencache internally to using an rcu locked hash table.